compiler/src/iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -1637,6 +1637,135 @@ Im2colOp::reifyResultShapes(OpBuilder &b,
       .reifyResultShapes(b, reifiedReturnShapes);
 }
 
+//===---------------------------------------------------------------------===//
+// Custom Op
+//===---------------------------------------------------------------------===//
+
+unsigned CustomOp::getNumLoops() { return getIteratorTypesAttr().size(); }
+
+int64_t CustomOp::getRank(Value v) {
+  Type type = v.getType();
+  if (type.isIntOrIndexOrFloat()) {
+    return 0;
+  }
+  return cast<RankedTensorType>(type).getRank();
+}
+
+LogicalResult CustomOp::verify() {
+  // All inputs/outputs must have indexing maps.
+  if (static_cast<int64_t>(getIndexingMapsAttr().size()) != getNumOperands()) {
+    return emitOpError("expected number of indexing maps (")
+           << getIndexingMapsAttr().size()
+           << ") to be same as the "
+              "number of input/output operands ("
+           << getNumOperands() << ")";
+  }
+
+  // Check the form of the indexing maps.
+  std::optional<unsigned> numSymbolDims;
+  for (auto [index, indexingMapAttr, operand] :
+       llvm::enumerate(getIndexingMapsAttr(), getOperands())) {
+    auto indexingMap = cast<AffineMapAttr>(indexingMapAttr).getValue();
+    if (indexingMap.isEmpty()) {
+      continue;
+    }
+
+    // Domain must be consistent.
+    unsigned numLoops = getNumLoops();
+    if (indexingMap.getNumDims() != numLoops) {
+      return emitOpError("expected indexing_map #")
+             << index << " to have " << numLoops
+             << " dim(s) to match the number of loops or be zero";
+    }
+
+    // Check that number of symbols is consistent.
+    if (numSymbolDims) {
+      if (indexingMap.getNumSymbols() != numSymbolDims.value()) {
```

Comment:
I am thinking that we should be able to drop this requirement. It's not clear to me what we'd be able to do with the symbols other than maybe reuse `tensor.dim` ops in dynamic cases. Fine to leave here for now because I see what you're going for.